### PR TITLE
Feature/profile refactoring: PR #2 Functional changes

### LINKED ISSUE #1571: Refactor profile
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a developer, I want to refactor the profiles according to #1495 to improve developer experience.

### Description

Currently the structure of the profile directory looks like the following:
```
├── db.cljs
├── edit
│   └── views.cljs
├── events.cljs
├── photo_capture
│   ├── styles.cljs
│   └── views.cljs
├── qr_code
│   ├── styles.cljs
│   └── views.cljs
├── styles.cljs
├── validations.cljs
└── views.cljs
```
According to #1495 it should be:
```
├── edit
│   └── screen.cljs
├── handlers.cljs
├── photo_capture
│   ├── screen.cljs
│   └── styles.cljs
├── qr_code
│   ├── screen.cljs
│   └── styles.cljs
├── screen.cljs
├── specs.cljs
├── styles.cljs
└── validations.cljs
```
1. add letsubs macro in views
2. apply code convention for views (#1495)
3. move nav/preload-data! to navigation.cljs
4. rewrite events using fx and cofx
5. apply code convention for events (#1495)
6. rewrite subs using re-frame best practice (db can be used only in extraction subscriptions)
7. use ns keys in app-db
8. write specs
9. write tests

### PR DESCRIPTION
**Pull request by yenda**
## Second step of refactoring guidelines (for #1571) 
 - use fx and cofx
 - use ns keys for events
 - moved some event logic that was in the view into the events
 - keep state locally in the drawer
 - no subscriptions
 - specs

## Steps to test:

play around with profile editing and contact profile to see if anything broke
- change status from drawer
- change name from drawer
- edit profile
- try username "wallet" (should be in red and if saved go back to previous name)
- change image profile

### CODE REVIEW

**Comment by yenda** [comment]
Once in a while when selecting a new image profile in the image picker, nothing happens and you are still there in the image picker. Trying to pick an image again leads to a red screen with the following log:

08-16 22:37:27.682  2248  2248 E image-crop-picker: Promise rejected. Illegal callback invocation from native module. This callback type only permits a single invocation from native code.
08-16 22:37:27.683  2248  2299 E ReactNativeJS: The callback openPicker() exists in module ImageCropPicker, but only one callback may be registered to a function in a native module.

Since this bug occurs rarely I cannot tell for sure if it is due to this PR or not.

**Comment by yenda** [comment]
Current odd behaviors that I didn't change in my profile view:
- touching status opens edit-my-profile view
- touching profile icon and name does not

Changed behavior:
- pressing save now returns to my-profile-view (before it was staying in edit-my-profile-view)
- wan't sure if there can be linebreaks in the status or not, before a linebreak would be removed and the status saved on hitting the virtual keyboard `enter/return` key

File: src/status_im/components/drawer/view.cljs [hunk 1]
```diff
@@ -78,21 +69,17 @@
                     :max-length          140
                     :accessibility-label id/drawer-status-input
                     :placeholder         placeholder
-                    :default-value       status
-                    :on-blur             #(do
```
Comment (flexsurfer) [inline src/status_im/components/drawer/view.cljs]: why did you delete these all RND work?

File: src/status_im/components/drawer/view.cljs [hunk 2]
```diff
@@ -78,21 +69,17 @@
...
-                                              (rf/dispatch [:set-in [:profile-edit :status] new-status])))}]
+                    :default-value       (:status account)
+                    :on-change-text      #(reset! status-text %)
+                    :on-end-editing      #(do (reset! status-edit? false)
+                                              (rf/dispatch [:my-profile/update-status @status-text]))}]
+
        [status-view/status-view {:style           (st/status-view (str/blank? status))
                                  :on-press        #(reset! status-edit? true)
                                  :number-of-lines 3
-                                 :status          (if (str/blank? status) placeholder status)}])]))
```
Comment (flexsurfer) [inline src/status_im/components/drawer/view.cljs]: what is the reason to use `(:status account)` twice instead of using symbol?

File: src/status_im/ui/screens/profile/events.cljs [hunk 3]
```diff
@@ -1,58 +1,80 @@
...
-   prepare-edit-profile
-   open-edit-profile))
+  (fn [_ _]
+    {:dispatch-n [[:navigate-to :chat console-chat-id]
+                  [:select-chat-input-command {:name "phone"}]]}))
+
+(handlers/register-handler-fx
+  :my-profile/update-picture
+  (fn [{:keys [db]} [this-event base64]]
+    (if base64
```
Comment (jeluard) [inline src/status_im/ui/screens/profile/events.cljs]: This one is weirdish as we do 2 completely different things based on `base64`.

Maybe move this logic upfront?

File: src/status_im/components/drawer/view.cljs [hunk 4]
```diff
@@ -49,26 +43,23 @@
...
-                            (rf/dispatch [:account-update {:name (utils/clean-text new-name)}])))}]]))
+   name-text  (r/atom nil)]
+  [view st/name-input-wrapper
+   [text-input
+    {:placeholder    placeholder
+     :style          (st/name-input-text (s/valid? ::profile.db/name @name-text))
+     :font           :medium
+     :default-value  (or @name-text (:name account))
+     :on-change-text #(reset! name-text %)
+     :on-end-editing #(if (s/valid? ::profile.db/name @name-text)
```
Comment (jeluard) [inline src/status_im/components/drawer/view.cljs]: It would be nice to start a conversation about when to use `specs` validation.

File: src/status_im/ui/screens/profile/events.cljs (hunk 3 above)
Comment (yenda) [reply src/status_im/ui/screens/profile/events.cljs]: maybe it is clearer if I rename base64 to picture ? When called without a picture this event opens the image-picker, when called with a picture it updates it.

File: src/status_im/components/drawer/view.cljs (hunk 4 above)
Comment (yenda) [reply src/status_im/components/drawer/view.cljs]: I agree here I didn't really change anything regarding validation it was done like this before

File: src/status_im/components/drawer/view.cljs (hunk 2 above)
Comment (yenda) [reply src/status_im/components/drawer/view.cljs]: leftover of experimentation it's fixed

File: src/status_im/ui/screens/profile/events.cljs (hunk 3 above)
Comment (jeluard) [reply src/status_im/ui/screens/profile/events.cljs]: Not sure that would help much. I see handlers as more atomic, as in they just do one thing (either update a image, or allow user to select one).
A matter of taste.

**Comment by yenda** [comment]
I would like a first review at this point. More changes can be done but they require to make some decisions I shouldn't make alone.

See issue #1633

File: src/status_im/components/drawer/view.cljs [hunk 5]
```diff
@@ -118,8 +111,8 @@
   (fn [_ row-id _]
     (when (< row-id (dec transactions-count))
       (list-item
-        ^{:key row-id}
-        [common/separator {} st/transactions-list-separator]))))
+       ^{:key row-id}
```
Comment (flexsurfer) [inline src/status_im/components/drawer/view.cljs]: identation

File: src/status_im/ui/screens/profile/events.cljs (hunk 3 above)
Comment (janherich) [reply src/status_im/ui/screens/profile/events.cljs]: I agree with @jeluard on splitting this into two distinct handlers. While it's neat that the `:my-profile/update` picture can cause effect which in turn dispatches the same event again with different args (to be honest, for me, this is the first time where having dispatch keyword in the event vector argument is of clear utility :) ), I think that having them separate is still easier to reason and easier to extend in the future.
But not a big thing, renaming `base64` to `base64-image` or `image-data` would be sufficient for me.

Comment (yenda) [reply src/status_im/ui/screens/profile/events.cljs]: I renamed it base64-image if that is enough for you

File: src/status_im/components/drawer/view.cljs (hunk 5 above)
Comment (yenda) [reply src/status_im/components/drawer/view.cljs]: thanks yet another symbol to add to my emacs config, I need to check later if there isn't a general solution for that I noticed you like to limit the width of the lines by breaking lines after the function name :)

**Comment by oskarth** [comment]
> [WIP] specs

@yenda is still a WIP or is it ready to be merged? Also, a short summary might be useful, with bullets like "removed event X cause we now keep name state locally" or something to that effect, mostly in case there are regressions.

File: src/status_im/components/drawer/view.cljs [hunk 6]
```diff
@@ -49,26 +43,25 @@
...
+                          (rf/dispatch [:account-update {:name (utils/clean-text @name-text)}])
+                          (reset! name-text previous-name))}]]))
 
 (defview status-input []
   [account      [:get-current-account]
    status-edit? (r/atom false)
    status-text  (r/atom nil)]
-  (let [status      (:status account)
-        placeholder (i18n/label :t/update-status)]
+  (let [placeholder (i18n/label :t/update-status)
```
Comment (rasom) [inline src/status_im/components/drawer/view.cljs]: let's align symbols in `let` forms

**Comment by asemiankevich** [comment]
Tested both iOS / Android 
**0.9.10-14-g9c913db5+
0.9.11d14-1429.142**

1. Change default name
2. Change default status
3. Edit user picture
4. Checked user can not use "wallet", "console" as username, also "@", "/" are excluded
5. Edit profile

There are several UX concerns that are not caused by current PR , which I have added to https://github.com/status-im/status-react/issues/1633
